### PR TITLE
feat: Venafi Enhanced Issuer config + chart docs update

### LINKED ISSUE #422: feat: Add venafi enhanced issuer to chart
Helm install doesn't have the required config for Venafi Enhanced Issuer. Add this so this information is present in TLSPK.

### PR DESCRIPTION
**Pull request by hawksight**
- Adds ability to read `VenafiIssuer` and `VenafiClusterIssuer` resources to RBAC
- Adds `VenafiIssuer` and `VenafiClusterIssuer` to config
- Correct Typo in Notes
- Add extra log check in notes section
- Make `ConfigMap` & `Secret` Volumes required.
- Update agent version to `v0.1.39` after updates from #307
- Correct Chart.yaml Application version as well.
- Documentation update to chart README, now matching content [here](https://platform.jetstack.io/documentation/installation/agent)
- Update version of `helm-docs` used to match [latest available](https://github.com/norwoodj/helm-docs/releases/tag/v1.11.0)
-  Edited comments in values file to clean up some of the auto generated docs

Closes #422 

Pic showing that using this version of the chart I do indeed have a `VenafiIssuer` in TLSPK, a.k.a JSS:
Screenshot from 2023-04-26 14-52-02

I've tested my install but would appreciate someone else giving it a run over as well.

### CODE REVIEW

File: deploy/charts/jetstack-agent/README.md
Comment (sitaramkm) [inline deploy/charts/jetstack-agent/README.md:3]: Can you update this to read TLS Protect for Kubernetes Agent

File: deploy/charts/jetstack-agent/README.md
Comment (sitaramkm) [inline deploy/charts/jetstack-agent/README.md:103]: recommendation - can we turn the registry path to a variable as most often this will be installed via a mirrored location

File: deploy/charts/jetstack-agent/README.md.gotmpl
Comment (sitaramkm) [inline deploy/charts/jetstack-agent/README.md.gotmpl:8]: The TLS Protect for Kubernetes agent Helm chart

**Review by sitaramkm — CHANGES_REQUESTED** [review]
Some quick comments for consideration

File: deploy/charts/jetstack-agent/README.md
Comment (SpectralHiss) [reply deploy/charts/jetstack-agent/README.md:26]: we do have a process built in enterprise builds just pending a final review and merge here https://github.com/jetstack/enterprise-builds/pull/104

**Review by tfadeyi — APPROVED** [review]
lgtm

File: deploy/charts/jetstack-agent/README.md
Comment (hawksight) [reply deploy/charts/jetstack-agent/README.md:26]: Thanks @SpectralHiss - yes @sitaramkm please see that thread regarding this issue. The current `0.1.0` version of the chart is there because Houssem was proving out his work on publishing the chart, which may have only went to the eu registry. Hopefully once that PR is merged, all future versions will go to both registries.

File: deploy/charts/jetstack-agent/README.md
Comment (hawksight) [reply deploy/charts/jetstack-agent/README.md:3]: I agree this should be done, but I think that change will be much larger than just the title. To the extent of this repository being renamed or even moved to be under Venafi Organisation.
I will follow up with product to see where this is on their roadmap.

File: deploy/charts/jetstack-agent/README.md.gotmpl
Comment (hawksight) [reply deploy/charts/jetstack-agent/README.md.gotmpl:8]: I think I caught this with my latest commit

**Comment by hawksight** [comment]
@sitaramkm - I made a couple tweaks to use `TLSPK_REGISTRY` env var for a configurable registry.
Most of the docs added are a copy of the [documentation](export TLSPK_REGISTRY="eu.gcr.io/jetstack-secure-enterprise") added in the last week or two. (So will need to make the same tweak over there too)

I very nearly just linked to the website but thought it was nice to have a copy with the chart so it was a complete package.

Regarding naming, lets keep that for a separate thread because the scope of that could be huge. I've added https://github.com/jetstack/jetstack-secure/issues/431 to be a catch all for renaming / branding discussions.

**Review by SpectralHiss — APPROVED** [review]
nice work

**Pull request merged**
